### DISCUSSION TITLE
Running two function on onIonChange in in react hooks form

### DISCUSSION THREAD

**Discussion by ratnabh** [react-hook-form/react-hook-form#2874] [Q&A]
I have been using react hooks form but have come in situation where i need to set the state and also call onChange method to get rid of error for which i have made a validation. Here is my code

  ```
  <Controller
                render={({ onChange, onBlur, value }) => (
                  <IonSelect
                    placeholder="Select a State"
                   
                    onIonChange={onChange}
                    onIonChange={(e:any)=>{
                      handleChange(
                        "selectedStateForBilling",
                        String(e.detail.value)
                      ); 
                    }}
                   
                  >
                    {state.states.map((item) => (
                      <IonSelectOption value={item} key={item}>
                        {item}
                      </IonSelectOption>
                    ))}
                  </IonSelect>
                )}
                control={control}
                name="billingState"
                defaultValue=""
                rules={{
                  required: true,
                }}
              />
```

So looking at this code you may see i want to actually run two onIonChange method one for updating a state variable and the other to update the "billingState" so that i could get rid of error required:true  (which is defined in my rules={{}})  is also required to throw error as for the first if users clicks on submit leaving this value empty he sees it. So is there a way i can run onChange and handleChange both in onIonChange or is there a way i can make a check of required=true for state varaible ? as that would also do

**Comment by bluebill1049** [comment]
```
onIonChange={(e:any)=>{
                    handleChange(
                      "selectedStateForBilling",
                      String(e.detail.value)
                    ); 
onChange(e);
                  }}
```

?

**Reply by ratnabh** [reply]
ahh thank you so much ...tried this way but forgot to pass event parameter to onChange().. my bad:) and also thanks for this awesome lib.
